src/platform/plugins/shared/workflows_execution_engine/server/execution_functions/setup_dependencies.test.ts

```diff
@@ -246,4 +251,49 @@ describe('setupDependencies', () => {
     expect(mockActionsPlugin.getUnsecuredActionsClient).not.toHaveBeenCalled();
     expect(ConnectorExecutor).toHaveBeenCalledWith(mockScopedActionsClient, true);
   });
+
+  describe('WorkflowGraph', () => {
+    it('should call fromWorkflowDefinition with correct workflow definition', async () => {
+      await setupDependencies(
+        workflowRunId,
+        spaceId,
+        mockActionsPlugin,
+        mockTaskManager,
+        mockEsClient,
+        mockLogger,
+        mockConfig,
+        mockWorkflowExecutionRepository,
+        mockStepExecutionRepository,
+        mockLogsRepository,
+        {} as CoreStart,
+        mockDependencies
+      );
+
+      expect(WorkflowGraph.fromWorkflowDefinition).toHaveBeenCalledWith(
+        mockWorkflowExecution.workflowDefinition,
+        expect.anything()
+      );
+    });
+
+    it('should call fromWorkflowDefinition with correct default settings', async () => {
+      await setupDependencies(
+        workflowRunId,
+        spaceId,
+        mockActionsPlugin,
+        mockTaskManager,
+        mockEsClient,
+        mockLogger,
+        mockConfig,
+        mockWorkflowExecutionRepository,
+        mockStepExecutionRepository,
+        mockLogsRepository,
+        {} as CoreStart,
+        mockDependencies
+      );
+
+      expect(WorkflowGraph.fromWorkflowDefinition).toHaveBeenCalledWith(expect.anything(), {
+        timeout: '6h',
+      });
```

Comment:
Hey, thanks for adding tests, but I’m not sure these here are adding much value, they mainly check that parameters are passed to the setup function, rather than testing any actual logic. 
The tests in `timeout_zone_graph.test.ts`, which indirectly cover `resolveWorkflowSettings` (the logic added in this PR), might be enough.

Also, a small nitpick: I noticed the `defaultSettings` parameter is a static constant object. Since it’s always the same value, maybe it could be defined inside the setup (or even in the `resolveWorklfowSettings` function) or imported from a shared location, instead of having to drill an extra parameter to always pass the same constant value. That might simplify things a bit.